### DISCUSSION TITLE
SSAS crushed when using DAX studio

### DISCUSSION THREAD

**Discussion by DobrinStefanov** [DaxStudio/DaxStudio#642] [Q&A] [answered]
I have a strange behaviour when using DAX studio on a client machine. Just after a few minutes playing with DAX Studio the SSAS instance is on 100% CPU and everything freeze. The CPU is not great not terrible 
![image](https://user-images.githubusercontent.com/85291740/120641210-02638f00-c47c-11eb-9e2d-320a17cfd8fd.png)
On the other side the model is 25MB and have 1 fact table (1,5mil rows) and 3 small dimensions. All the measures are simple calculations like count, sum or simple calculates. 
Does anyone had such an issue. This is the first time such an issue occured. On every other machine where i worked with DAX studio everything was normal (yes it consume a lot of CPU when runing a queries but never stuck) Any suggestions will be more than welcomed.
![image](https://user-images.githubusercontent.com/85291740/120641786-b82edd80-c47c-11eb-828a-7eac76648901.png)

**Comment by dgosbell** [comment, marked as answer]
The only interaction DAX Studio does in the background without specific user interaction are metadata operations. When you alt-tab or click away from DAX Studio we check the last schema update date and if it is newer than the current metadata we will refresh the metadata pane. But this will not cause such excessive CPU and memory usage. Your screenshot is also using over 3Gb of RAM so something major is going on if your model is only 25mb.

Can you give us more information about what you mean by "playing"?

I assume this means you were running queries of some sort. Are you the only user on this machine? Are there other models on this SSAS instance? For the size of model you describe it would only be something like a processing operation or some sort of large crossjoin query that I would have expected to generate CPU load like that. But the memory usage is also way too high for a processing a 25mb model. 

The query cancelling mechanism is SSAS is cooperative. I have seen some rare bugs in the SSAS engine in the past where a query will get stuck in a loop or long running operation and cannot be cancelled. Then I've seen users try and cancel then re-issue these queries which ends up with multiple queries running consuming large amounts of CPU and RAM. It's pretty rare, but maybe something you did triggered this sort of bug.

**Reply by DobrinStefanov** [reply]
Thanks for the quick answer. And yes mayble i can give you a little bit more details. 
On the server there are 3 more cubes but they are not scheduled for processing and when this occures the cubes were idle (how big they are i do not know , i can check but if they are idle i think they are irrelevant for us at this point). 
By playing i mean run a simple Evaluate queries like Summarizecolumn ( dim_address[Country], "Count" , [Counnt active] ) where [count active] is a measure that just count the active sales shops ( as simple count over the fact table (1,5mil rows) ) or run the metrics to check what is the volume of the Dictionary and Cardinarity. As i said nothing really heavy. And due to client's honest word I am the only one on the server and nobody is doing nothing. As i am not familiar with  what is going in the background between SSAS and DAX studio my asumption is that just the Server is very old. As i said i have never expirienced such a behaviour with DAX Studio on any other machines. It is not a big deal to have one project without DAX studio but if i hit a performace issue i would like to have the DAX studio availble so i can investigate what is going on.

**Reply by dgosbell** [reply]
I have not seen anything in your description that indicates that there is any DAX Studio issue here. 

If you run the same query multiple times do you observe the same behaviour? 

What do the Server Timings look like for this query? can you post a screen shot?

Have you tried running the same query via another tool like SSMS? And if so is the behaviour any different?
